### PR TITLE
Refactor translator into react component

### PR DESCRIPTION
**Pull request by shifucun**
Plan to have a more realistic playground for external users to use with their own Bigquery Dataset. This would pave the path for that.

### CODE REVIEW

File: static/js/translator/constant.ts [hunk 1]
```diff
@@ -0,0 +1,298 @@
+/**
```
Comment (beets) [inline static/js/translator/constant.ts:1]: nit: constants.ts

File: static/js/translator/page.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
+/**
```
Comment (beets) [inline static/js/translator/page.tsx:1]: nit: please keep each file to one component

**Review by beets — COMMENTED** [review]
curious about the path to that -- would we have a mixer api to any bigquery?

left small comments, mostly style nits.

**Review by shifucun — COMMENTED** [review]
Updated

File: static/js/translator/page.tsx (hunk 2 above)
Comment (shifucun) [reply static/js/translator/page.tsx:1]: Done

**Review by beets — APPROVED** [review]
thanks for making the changes!

**Pull request merged**